android/app/src/main/java/com/androidstartupperfapp/navigation/initialroutenamestate/InitialRouteNameState.java

```diff
@@ -0,0 +1,41 @@
+package com.androidstartupperfapp.navigation.initialroutenamestate;
+
+public final class InitialRouteNameState {
+
+  //region Value
+  private static String value;
+
+  public static String getValue() {
+    return value;
+  }
+
+  public static void setValue(String value) {
+    InitialRouteNameState.value = value;
+    InitialRouteNameState.onChange();
+  }
+  //endregion
+
+  //region Value change event emitter
+  public interface OnChangeListener {
+    void onChange(String initialRouteName);
+  }
+
+  private static OnChangeListener onChangeListener;
+
+  public static OnChangeListener getOnChangeListener() {
+    return onChangeListener;
+  }
+
+  public static void setOnChangeListener(OnChangeListener onChangeListener) {
+    InitialRouteNameState.onChangeListener = onChangeListener;
+  }
+
+  private static void onChange() {
+    if (onChangeListener == null) {
+      return;
+    }
+
+    onChangeListener.onChange(value);
+  }
+  //endregion
+}
```

Comment:
# Sharing state between Android and React

## React Native bridge 
This design pattern could be reused in a lot of cases, especially for simpler UI scenarioss. Not just initial route name state.

Take "`username` & `password` input" for example.

For React to know about the `username` & `password` values, all it needs to have is a way to retrieve them from running memory in the Android layer. A custom React Native bridge should solve this easily:

- get and set methods will be the mainstay solution to retrieve and update the shared values
- event listeners if the situation calls for responding to live changes to the shared values

## Android Shared Preferences (...which also requires React Native bridge)
(going back to the "`username` & `password` input" scenario...)

An alternative considered is to use a more powerful, centralized disk-based storage ([shared preferences](https://developer.android.com/reference/android/content/SharedPreferences.html) and [its encrypted counterpart](https://developer.android.com/reference/kotlin/androidx/security/crypto/EncryptedSharedPreferences)), in the same way one would do with Redux.

This could be the way to go when the information needs to persist when the app is not alive.

For short-term memory manipulation, though, it's overkill: the app has to go through additional steps and setup to store and parse the value. This goes double for when the value is sensitive information. By using this approach, the app risks:

- an additional point of failure (is storage unavailable?)
- an additional vector of attack (is the storage exposed?)

Careful consideration must be done before this approach is taken.